services/bettersocial/fcmToken/sendGroupChatNotification.js

```diff
@@ -10,12 +10,13 @@ const sendGroupChatNotification = async (userTargetId, message) => {
       title: message,
       body: message
     },
-    data: {}
+    data: {},
+    token: userTargetToken?.token
   };
   if (userTargetToken) {
     try {
       messaging()
-        .sendToDevice(userTargetToken?.token, payload)
+        .send(payload)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code is using the `send` method instead of `sendToDevice`. However, it seems like the payload is being sent even when `userTargetToken` is undefined due to the placement of the token assignment in the payload. This could lead to errors or failed sends. I suggest moving the token assignment inside the if condition to ensure that the payload is only sent when the token exists.

```diff
-    data: {},
-    token: userTargetToken?.token
+    data: {}
   };
   if (userTargetToken) {
+    payload.token = userTargetToken.token;
     try {
       messaging()
        .send(payload)
         .then(() => {});
     } catch (error) {
       console.log('error send group chat notifications', error);
```
Also, consider handling the promise returned by `messaging().send(payload)` properly. Currently, it's just logging an error and not doing anything else. You might want to add some error handling logic there.


<!-- This is an auto-generated comment by OSS CodeRabbit -->